### PR TITLE
[COMDLG32] Enable saving files with different extension

### PR DESCRIPTION
**Pull request by zecarlos1957**
[COMDLG32] Enable saving files with different extension

like on Windows, gives precedence to the user's extension in file names if there is one.
Otherwise appends the selected extension. If not selected, append default extension.

### CODE REVIEW

**Comment by JoachimHenze** [comment]
Will saving something as "hello" still save it as "hello.txt" after your change?

**Comment by zecarlos1957** [comment]
@JoachimHenze 
If you want a file without extension, you should change the file types  combo box to (*.*) mask (Like Windows).
Now, with my changes notepad has the same behavior as Windows.

**Review by JoachimHenze — CHANGES_REQUESTED** [review]
I disagree: I just tested on XPSP3: If I do select "*.*" as the mask and then save a file as "hello", the file will get saved as "hello.txt", not as "hello".
In other words: I interpret this result as: We are required to attach ".txt" whenever no extension was added explicitly in the editbox. Even if "*.*" is the current mask.

**Comment by zecarlos1957** [comment]
@JoachimHenze 
Please open the command line in WinXP in the directory where you saved the hellow file, type the DIR command and check the file name. Probably the explorer has the hidden extensions. I await your confirmation.

**Comment by JoachimHenze** [comment]
No this is not the cause for the result I saw with MS notepad on XPSP3.
For 2 reasons:
1.) My OS is always set up to never hide any extensions.
2.) Hiding the extensions would only hide extension, but would not add extensions.

**Comment by zecarlos1957** [comment]
you right

**Comment by zecarlos1957** [comment]
@JoachimHenze  your comment are addressed.
 Thanks for your review

File: dll/win32/comdlg32/filedlg.c [hunk 1]
```diff
@@ -3096,9 +3096,10 @@ BOOL FILEDLG95_OnOpen(HWND hwnd)
 	else
 	  fodInfos->ofnInfos->Flags &= ~OFN_READONLY;
 
-        /* Attach the file extension with file name*/
+        /* if the user don't supply a extension, attach default extension to file name*/
```
Comment (binarymaster) [inline dll/win32/comdlg32/filedlg.c:3099]: This is Wine code. If you want to modify only comment here, it's better to do inside a `__REACTOS__` guard below.

File: dll/win32/comdlg32/filedlg.c [hunk 2]
```diff
@@ -3096,9 +3096,10 @@ BOOL FILEDLG95_OnOpen(HWND hwnd)
...
         ext = PathFindExtensionW(lpstrPathAndFile);
 #ifdef __REACTOS__
+         if (! *ext)
```
Comment (binarymaster) [inline dll/win32/comdlg32/filedlg.c:3102]:
```suggestion
        if (*ext == UNICODE_NULL)
```

File: dll/win32/comdlg32/filedlg.c (hunk 1 above)
Comment (zecarlos1957) [reply dll/win32/comdlg32/filedlg.c:3099]: yes

**Comment by zecarlos1957** [comment]
@JoachimHenze 
Yours requests are addressed. Can you review please ?
https://github.com/reactos/reactos/pull/4903

File: dll/win32/comdlg32/filedlg.c [hunk 3]
```diff
@@ -3099,6 +3099,7 @@ BOOL FILEDLG95_OnOpen(HWND hwnd)
         /* Attach the file extension with file name*/
         ext = PathFindExtensionW(lpstrPathAndFile);
 #ifdef __REACTOS__
+        if (*ext == UNICODE_NULL)
```
Comment (learn-more) [inline dll/win32/comdlg32/filedlg.c:3102]: Should this also take into account `fodInfos->defext` like the original code does?

**Review by JoachimHenze — APPROVED** [review]
Approved. We are within a ReactOS block here, so we are free to experiment.
Ftr whenever you explicitly want to (re)request from me a review for any of your PRs, it is best to contact me in chat.reactos.org. Reaching me via email or github is much more likely to miss it.

File: dll/win32/comdlg32/filedlg.c (hunk 3 above)
Comment (zecarlos1957) [reply dll/win32/comdlg32/filedlg.c]: Yes it should. Thanks.

**Pull request merged**